src/components/SummaryBar/SummaryBar.jsx

```diff
@@ -170,11 +175,26 @@ const SummaryBar = props => {
     }
   };
 
+  const authenticateUserRole = authenticateUser ? authenticateUser.role : '';
+
   if (userProfile !== undefined && leaderData !== undefined) {
     const weeklyCommittedHours = userProfile.weeklycommittedHours + (userProfile.missedHours ?? 0);
     const weeklySummary = getWeeklySummary(userProfile);
     return (
-      <Container fluid className={matchUser ? 'px-lg-0 bg--bar' : 'px-lg-0 bg--bar disabled-bar'}>
+      <Container
+        fluid
+        className={
+          matchUser ||
+          hasPermission(
+            authenticateUserRole,
+            'submitWeeklySummaryForOthers',
+            roles,
+            authenticateUserPermission,
+          )
+            ? 'px-lg-0 bg--bar'
+            : 'px-lg-0 bg--bar disabled-bar'
```

Comment:
I suggest moving this logic up before the return, assigning it to a const, and passing that to className for improved readability.